plans/upgrade.pp

```diff
@@ -114,21 +116,18 @@
     )
   }
 
-  # Shut down Puppet on all infra targets
-  run_task('service', $all_targets,
-    action => 'stop',
-    name   => 'puppet',
-  )
+  # Shut down Puppet on all infra targets. Avoid using the built-in service
+  # task for idempotency reasons. When the orchestrator has been upgraded but
+  # not all pxp-agents have, the built-in service task does not work over pcp.
+  run_command('systemctl stop puppet', $all_targets)
 
   ###########################################################################
   # UPGRADE MASTER SIDE
   ###########################################################################
 
-  # Shut down PuppetDB on CMs that use the PM's PDB PG
-  run_task('service', $compiler_m1_targets,
-    action => 'stop',
-    name   => 'pe-puppetdb',
-  )
+  # Shut down PuppetDB on CMs that use the PM's PDB PG. Use run_command instead
+  # of run_task(service, ...) so that upgrading from 2018.1 works over PCP.
+  run_command('systemctl stop pe-puppetdb', $compiler_m1_targets)
```

Comment:
This assumes systemctl is available on the host.  Since 2018 supports RHEL6 this wouldn't work in all cases.

https://puppet.com/docs/pe/2018.1/supported_operating_systems.html#supported_operating_systems